lesson16/main.js

```diff
@@ -0,0 +1,111 @@
+// json = https://mocki.io/v1/d3176c84-479a-47a3-a8b6-7d597b55d8b4
+const ul = document.getElementById("js-ul");
+const li = document.createElement("li");
+const contetns = document.getElementsByClassName("tab-content");
+const tabs = document.getElementsByClassName("tab-item");
+
+const displayInfo = (info) => {
+  const li = document.createElement("li");
+  if (typeof info === "string") {
+    li.textContent = info;
+    ul.appendChild(li);
+    return console.log(info);
+  }
+  throw new Error("Please pass a string value");
+};
+
+const displayErrorInfo = (error) => {
+  console.log(error);
+  li.textContent = `Failed to get data from server`;
+  ul.appendChild(li);
+};
+
+const getData = async (api) => {
+  let response = null;
+  try {
+    response = await fetch(api);
+  } catch (error) {
+    displayErrorInfo(error);
+  }
+  if (response.ok) {
+    return response.json();
+  } else {
+    displayErrorInfo(response);
+  }
+};
+
+const checkResponse = async (callBackFunction) => {
+  let response = await getData(
+    "https://mocki.io/v1/40cbd0f7-2a64-4b95-a991-57ff982901a8"
+  );
+  if (response.data) {
+    callBackFunction(response.data);
+  } else {
+    displayInfo("no data available");
+  }
+};
+
+const createTabs = (items) => {
+  ul.classList = "tab-menu";
+  const fragment = new DocumentFragment();
+  for (let item of items) {
+    const li = document.createElement("li");
+    li.textContent = item.category;
+    li.classList = "tab-item";
+    fragment.appendChild(li);
+  }
+  ul.appendChild(fragment);
+  activateFirstTab();
+};
+
+const activateFirstTab = () => {
+  tabs[0].classList.add("active");
+};
+
+// this function will be modified to render articles, comments, an image and icons.
+const createArticles = (items) => {
+  const fragment = new DocumentFragment();
+  const articleContainerDiv = document.createElement("div");
+  articleContainerDiv.classList.add("tab-box");
+  for (let item of items) {
+    const article = document.createElement("div");
+    article.classList.add("tab-content");
+    article.textContent = item.category;
+    fragment.appendChild(article);
+  }
+  articleContainerDiv.appendChild(fragment);
+  ul.insertAdjacentElement("afterend", articleContainerDiv);
+  activateFirstArticleContent();
+};
```

Comment:
IMO
I was curious about the naming.
I thought it would be more intuitive to use ``newsData``  or somthing instead of ``item``.
I also thought that a ``tab-content-box`` would be better than a ``tab-box``.
I have been pointed out before but ``render`` is more appropriate more than ``create``.

```suggestion
const renderArticlesContents = (newsDatas) => {
  const fragment = new DocumentFragment();
  const articleContainerDiv = document.createElement("div");
  articleContainerDiv.classList.add("tab-content-box");
  for (let newsData of newsDatas) {
    const article = document.createElement("div");
    article.classList.add("tab-content");
    article.textContent = newsData.category;
    fragment.appendChild(article);
  }
  articleContainerDiv.appendChild(fragment);
  ul.insertAdjacentElement("afterend", articleContainerDiv);
  activateFirstArticleContent();
};
```

Reply:
Thank you for your suggestion 
I updated the name of the functions and parameters!
Please see the latest commit 
59fb045fc215cf1bab0148169691a59cf80422a1